### PR TITLE
[FLINT] new recipe

### PR DESCRIPTION
**Pull request by benlorenz**
FLINT: Fast Library for Number Theory: http://www.flintlib.org/

Needed for oscar-system and related software (Nemo.jl, Oscar.jl, Polymake.jl).
Script adapted from Flint2Builder by @thofma but with support for all platforms.

cc: @wbhart

### CODE REVIEW

**Comment by thofma** [comment]
Amazing, thanks @benlorenz! I guess you noticed the millions of warnings in the log for the Windows binaries (the logs were bigger than the binaries). When I looked at this together with @tthsqe12, we came to the conclusion that one should remove the `-ansi` from the `$CFLAGS` at https://github.com/wbhart/flint2/blob/trunk/configure#L581. This was on my todo list, but I did not have time to implement it.

File: F/FLINT/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+elif [[ ${target} == *mingw* ]]; then
+   # fix arch detection:
+   sed -i -e 's/$(ARCH)/$ARCH/g' configure
+   extraflags=--reentrant
+   libsubdir=bin
+fi
+
+./configure --prefix=$prefix --disable-static --enable-shared --with-gmp=$prefix --with-mpfr=$prefix ${extraflags}
+make -j${nproc}
+make install LIBDIR=$libsubdir
```
Comment (giordano) [inline F/FLINT/build_tarballs.jl]: We already have the [environment variable](https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Automatic-environment-variables-1) for this, no need to manually set it (remove also above)
```suggestion
make install LIBDIR=${libdir}
```

Comment (benlorenz) [reply F/FLINT/build_tarballs.jl]: `$libdir` is not correct here as the Makefile expects this to be just the `lib` or `bin` but `$(basename ${libdir})` should do and is nicer than setting it manually.

**Comment by tthsqe12** [comment]
@thofma how hard would be to offer flint/arb without the -g? a 30+MB dll is obscene.

**Comment by wbhart** [comment]
We need the -g for the stack traces in Julia.

**Comment by tthsqe12** [comment]
I know. But, if I don't want obscenely large binaries, must I build them myself?

**Comment by thofma** [comment]
Yes, probably.

**Comment by tthsqe12** [comment]
ok.

**Comment by wbhart** [comment]
You could fork the builder on GitHub and remove the -g. Of course you need to set up a GitHub deployment key and make a release of your builder in order to use it. Then you'd need to change the builder used in deps/build.jl in your Nemo.

**Comment by fingolfin** [comment]
what exactly is the problem with the "obscenely large binary", though? @tthsqe12 we can buy you a new SSD if you are running low on storage ;-)

**Pull request merged**